test/AutoDiff/SILOptimizer/optional.swift

```diff
@@ -0,0 +1,75 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import DifferentiationUnittest
+
+var OptionalTests = TestSuite("OptionalDifferentiation")
+
+//===----------------------------------------------------------------------===//
+// Basic tests.
+//===----------------------------------------------------------------------===//
+
+/*
+// Note: this lowers to `try_apply` instead of `switch_enum`, and is
```

Comment:
Please reference TF-433, which tracks `try_apply` differentiation:

```suggestion
// TODO(TF-433): this lowers to `try_apply` instead of `switch_enum`, and is
```